### ISSUE TITLE
English version of HgbThesisTutorial

### ISSUE THREAD

**Issue by hochleitner** [Digital-Media/HagenbergThesis#140]
`HgbThesisTutorial` needs to be translated into English to provide a support document for non-German-speaking students.

This is a larger effort and will be handled in the branch [`hgbthesistutorial-en-translation`](https://github.com/Digital-Media/HagenbergThesis/tree/hgbthesistutorial-en-translation).

Use this task list to mark things done and add a comment when working on one of the tasks. Please extend the list if tasks are missing.

- [x] (WH) Rename `HgbThesisTutorial` to `HgbThesisTutorialDE`
- [x] (WH) Make a copy of `HgbThesisTutorialDE` and call it `HgbThesisTutorialEN`. All translation tasks will happen in this directory.
- [x] (WH) Translate contents of `main.tex` (use comments/instructions from other English documents) and switch to `english`.
- [x] (WH) Translate `front/vorwort.tex`. Rename the file.
- [x] (WH) Adapt `front/kurzfassung.tex` (put in a `german` environment).
- [x] (WH) Adapt `front/abstract.tex` (remove `english` environment).
- [x] (WH) Translate `chapters/einleitung.tex`. Rename the file.
- [x] (WH) Translate `chapters/abschlussarbeit.tex`. Rename the file.
- [x] (WH) Translate `chapters/latex.tex`.
- [x] (WB) Translate `chapters/abbildungen.tex`. Rename the file.
- [x] (WB) Translate `chapters/mathematik.tex`. Rename the file.
- [x] (WH) Translate `chapters/literatur.tex`. Rename the file.
- [x] (WB) Translate `chapters/drucken.tex`. Rename the file.
- [x] (WB) Translate `chapters/schluss.tex`. Rename the file.
- [x] (WB) Translate `back/anhang_a.tex`. Rename the file.
- [x] (WB) Translate `back/anhang_b.tex`. Rename the file.
- [x] (WB) Translate `back/anhang_c.tex`. Rename the file.
- [x] (WB)Translate `back/anhang_d.tex`. Rename the file.
- [x] (WB) Translate `back/messbox.tex`. Rename the file.
- [x] (WH) Add information on headline capitalization (https://capitalizemytitle.com/) and that references to chapters, sections, figures, etc. should be capitalized (e.g., "see Section 3.4", "as shown in Figure 2.1").
- [x] (WH) Check if entries in `references.bib` need an update or if new ones need to be added for the English version.
- [x] (WH) Update `README.md`. Update the description and the table of documents.
- [x] (WH) Update `CHANGELOG.md`. Add translation as a new addition.

**Comment by hochleitner** [comment]
@imagingbook: I'm soon going to start the last big task for the "Release 2023" milestone - the English translation of `HgbThesisTutorial`. Since this is an enormous task, working on it locally and creating the PR seems not very practical. I, therefore, made and pushed a branch where I'll continuously work. You're welcome to join in on the fun, but no worries if you want to sit this out.
If you want to help, leave a comment on this issue, stating what you're currently working on so that we don't work on the same file simultaneously. Once something is finished, tick it off in the task list.
I have some grading to do this week, so I might not get to start before next week.

**Comment by imagingbook** [comment]
Sure I can help in this monumental task.
Could we perhaps add another tickbox to each item to mark if it is being worked on? And by whom?

Do you want to simultaneously re-edit the German version as well?

**Comment by hochleitner** [comment]
I don't think these task lists support more than one checkbox per line. That's why I felt we just wrote a comment, "I'm working on task X". Or, you could edit my issue message and add your name behind a task, marking it as a work in progress.

As for the German version, yes, that makes the most sense. We should also do that if we come across something while translating that needs to be updated in the German version.

**Comment by imagingbook** [comment]
Very well, but I'm afraid this could get confusing. How about simply marking the tasks in your original list?, like

- [ ] (WH) Rename `HgbThesisTutorial` to `HgbThesisTutorialDE`

etc.

**Comment by hochleitner** [comment]
Yes, that's what I meant with my second suggestion. I put us on the waitlist for the new tasklist feature (https://docs.github.com/en/issues/tracking-your-work-with-issues/about-tasklists) together with a new Roadmap feature. We got accepted for the latter but not the new task lists. If that happens in time, we will see if they are more flexible.

Meanwhile, let's do it as suggested above. I should be done with grading my 160+ exercises on Friday, so I might start on the weekend.

**Comment by imagingbook** [comment]
Ah, you are so smart ...

**Comment by hochleitner** [comment]
Okay, I'm getting this party started. I'll begin with renaming the folders, copying the German version to a second folder, and renaming the main file and the included chapters. Once all that is done, we can begin translating the actual content. I'll report back when I'm done.

**Comment by hochleitner** [comment]
Okay, done. The English version does exists; files are renamed to the filenames already used in `HgbThesisEN`, and `main.tex` has been adapted. We can now start translating the chapters. I'll go ahead and start with the three files in the `front` directory.

**Comment by hochleitner** [comment]
Okay, the `front` chapters are done. There are a few references to sections further back. I've translated the labels but nothing more so far. We still have to decide if we keep everything. For example, there's a section on writing your thesis in English. I don't really think we need this vice versa because no one in an English-speaking program will think about writing in German. Or it isn't even possible.

**Comment by imagingbook** [comment]
Excellent! No, we should not keep the whole material, this is a good time for a cleanup.
(Sorry, I'm currently too busy to work on this myself ...)

**Comment by imagingbook** [comment]
@hochleitner I just completed the translation to `figures.tex`, pls. have a look. Some labels in other files still need updating.

I am somewhat unhappy about the `longtable` example (Tab. 4.3). It looks OK in the German version but breaks the page strangely in the English version, thereby leaving *a lot* of empty space. I understand there is no easy way to enforce better placement of `longtable` . I am worried this beast could create more trouble than do any good...

**Comment by hochleitner** [comment]
Thanks for the effort! I'm done with `latex.tex`. That was a huge chapter... phew. I updated the outdated labels; everything should work again without warnings.

Concerning the long table: If you dislike it, we can always throw it out and leave the code-listing. Alternatively, we can create a wiki entry with an Overleaf example document. I'm also okay with that, but the students need some form of working example since long tables are being asked for every year.

**Comment by imagingbook** [comment]
OK, I revised the longtable description and also modified the example - pls. check! For this purpose, I also added two new items to `hgb.sty`:
* A `block` (dummy) environment to provide a limited scope for variable redefinitions (such as lengths). This eliminates the `{...}` brackets around environments, which are easily dropped and may cause hard to find errors.
* A `\getcurrentlabel` macro to retrieve the most recent label (used in the headers of follow-up pages in long tables).

I also replaced all legacy TeX `\def` commands by `renewcommand`. I think this should be done in other files too.

**Comment by imagingbook** [comment]
Mathematics chapter is done. I also added some info on available fonts in math mode (both in DE/EN).

**Comment by imagingbook** [comment]
Printing chapter is done. 
* Omitted the final section on providing a CD/DVD since this is obsolete. Consider adding a substitute (online resources)
* Added package `lengthconvert` to `hgb.sty` for showing the current textwidth in mm (found no easy other solution).

**Comment by hochleitner** [comment]
I got stuck in the middle of the literature chapter because I got sick. Will still need a few days to recover and will continue then. Thanks for the work so far!

**Comment by hochleitner** [comment]
Okay, this took longer than I hoped for, but it's done now. So if I'm correct, we should have translated everything now. Thanks for the effort!
I already reviewed everything quickly, corrected broken references, and switched some en dashes to em dashes. I'll also add a short text about title capitalization.
References look good so far. I don't think that anything needs to be changed or added. I might go over it again to see if any manuals have been updated in the meantime.

Is there anything that's still open on your part?

**Comment by imagingbook** [comment]
Very good. I checked the literature chapter again, fixed some minor errors/omissions and wording. Pls. look over it once more.

One thing I did in this and other chapters is to remove the use of boldface type wherever possible (since we recommend against it in the tutorial). Some other chapters still need to be checked ...

**Comment by hochleitner** [comment]
I wrote short introductory sentences for the citations sections where you left a comment.

I also added a subsection about title capitalization in `latex.tex`, section 3.3.2 (Headings). I think it fits best there, but please change it if you feel it should be somewhere else.

Also, in Section 3.3.6 (Cross-References), I added that terms like "Section", "Chapter", etc. should be capitalized. Please also check.

I updated `README.md` to reflect the new document structure and sorted the table so that the English version is always before the German one. I think it fits better since the whole project and README are in English.

I'll review the whole English document once more to check for typos and formatting.

I'll also have a look at the German one. The reformatting sometimes created issues with listings since the code is indented like this:

```latex
\begin{JavaCode}
     int number = 1;
    ...
\end{JavaCode}
```

This leads to the whole code being shifted to the right, which doesn't look good and makes no sense. We need to reformat the listings like this:

```latex
\begin{JavaCode}
int number = 1;
...
\end{JavaCode}
```

**Comment by hochleitner** [comment]
I proofread both language versions and made some adaptions. I hope that most of the issues should be gone. Yet I'm sure we'll still find enough over time.

Concerning the boldface: I agree but didn't touch these yet. What did you use to replace it? `\emph{}`? It's okay for unordered lists and things before a colon, isn't it?

If you want to replace these, let me know, or please go ahead. Other than that, the CHANGELOG needs to be updated when we're done, and then we can create a PR after a complete rebuild.

**Comment by imagingbook** [comment]
I did another round of revisions, mainly

* removed most `\textbf`uses (mostly by `\emph`, I think it is OK to leave it in places that are clearly "instructional", so people don't feel encouraged to use it in their own text),
* replaced many `verbatim` blocks by code listings,
* rephrased some english sentences for better clarity,
* plus many more small corrections.

One more thing: Perhaps we should shorten the Preface/Vorwort by removing some of the historic storytelling. Instead, I would like to see a paragraph re-stating the main purpose  (and benefits) of this package, along these lines: 

* LaTeX is great but gets messy very easily when things don't work as expected.
* We want to spare the "innocent" user from such trouble but nevertheless provide a professional authoring environment with a rich and proven functionality.
* It works transparently, such that authors do not need to deal with the inticate details underneath.
* The package does most of what is ever needed, authors do not need to go out and search for additional packages, worry about compatibility, fiddle with integration, ... 
* Simply put, they do not need to waste their time on technical details but can focus on creating content without compromising on the quality of the results.
* Besides, this template provides some really sophisticated functions in a well-balanced setup, not easily found elsewhere. 
* etc.

Didn't have time to formulate this but perhaps someone else can try ...

**Comment by hochleitner** [comment]
Sure, good idea. Would you like it in this release already? I can try and come up with something in the next few days.

Another thing I would like to know is what we should do while proofreading. The `algorithmx` package generates nice pseudocode that is also colored/highlighted in a way that makes it more understandable and readable. I'm getting more and more convinced that we should enable syntax highlighting for our program listings. We don't need to do this right now, but it's something that we can try out in a separate branch over time and see if we come to like it. We have some pretty long code fragements that look like a big gray wall of text; maybe we can improve that.

**Comment by imagingbook** [comment]
> Sure, good idea. Would you like it in this release already? I can try and come up with something in the next few days.

No, there is no need to press this into this release. Let's just keep it on the agenda...

> Another thing I would like to know is what we should do while proofreading. The `algorithmx` package generates nice pseudocode that is also colored/highlighted ...

Yes, the new algorithm setup looks good (I used it extensively in our last book edition). In general, a wider use of color is something we should look into, including syntax highlighting, but also section headings etc. But this can also wait till the next release.

So, are we ready to build and upload to CTAN yet?

**Comment by hochleitner** [comment]
> So, are we ready to build and upload to CTAN yet?

Yes, I'd say so. I'll do a full rebuild and create the PR right away.
